### PR TITLE
Linux

### PR DESCRIPTION
**Pull request by miguelangel-dev**
## Goal

This PR makes compatible Bow OpenAPI binary with Unix systems:
- macOS
- `Linux`

Also, this PR migrate from `Travis` to `Github Actions` our CI.

Screen Shot 2020-05-04 at 15 58 05

## Testing details

You can generate the whole module-tests compatible with Linux, running from macOS:
```bash
$ swift test --generate-linuxmain
```

After, go to Linux and using this auto-generated tests code, run (in root folder):
```bash
$ swift test
```

Screen Shot 2020-05-04 at 17 21 06

## References
- [FoundationNetworking module](https://github.com/apple/swift-corelibs-foundation/pull/2289)

### CODE REVIEW

File: Documentation.app/Contents/MacOS/Quick start.playground/Pages/Installation guide.xcplaygroundpage/Contents.swift [hunk 1]
```diff
@@ -6,15 +6,35 @@
...
+ | grep -oP '"tag_name": "\K(.*)(?=")' \
+ | xargs -I {} wget -O - https://github.com/bow-swift/bow-openapi/archive/{}.tar.gz \
+ | tar xz \
+ && cd bow-openapi-* \
+ && sudo make linux
+ ```
+
+ > It will install the last stable version.
+
+ Bow OpenAPI depends on `Java 8+` and `swagger-codegen`; both will be installed from `Makefile`. After the installation, you can type `bow-openapi -h` in a new termintal to check it.
```
Comment (truizlop) [inline Documentation.app/Contents/MacOS/Quick start.playground/Pages/Installation guide.xcplaygroundpage/Contents.swift]:
```suggestion
 Bow OpenAPI depends on `Java 8+` and `swagger-codegen`; both will be installed from `Makefile`. After the installation, you can type `bow-openapi -h` in a new terminal to use it.
```

File: Tests/APIHelperTests.swift [hunk 2]
```diff
@@ -2,36 +2,40 @@
...
-            return removed == expected
+            return expected == removed
         }
         
         property("Only items with nil values generate an empty URLQueryItem") <- forAll(self.nonePresentGen) { (absent) in
-            return absent.any.toQueryItems == nil
+            absent.any.toQueryItems == nil
+        }
+        
+        property("Items with none nil values generate valid URLQueryItems") <- forAll(self.allPresentGen) { (present) in
```
Comment (truizlop) [inline Tests/APIHelperTests.swift]:
```suggestion
        property("Items with no nil values generate valid URLQueryItems") <- forAll(self.allPresentGen) { (present) in
```

File: Tests/APIHelperTests.swift [hunk 3]
```diff
@@ -2,36 +2,40 @@
...
 import XCTest
 import SwiftCheck
+@testable import FixturesAPI
 
 
 class APIHelperTests: XCTestCase {
-    let allPresentGen: Gen<[String: String?]>  = [String: String].arbitrary.map { dict in dict.mapValues { x in x as String? } }
-    let nonePresentGen: Gen<[String: String?]> = [String: String?].arbitrary.map { dict in dict.mapValues { _ -> String? in nil } }
+    let allPresentGen: Gen<[String: String?]>  = [String: String].arbitrary.map { dict in dict.map { (k, v) in ["\(k)-present": v as String?] }.combineAll() }
+    let nonePresentGen: Gen<[String: String?]> = [String: String?].arbitrary.map { dict in dict.map { (k, _) in ["\(k)-none": nil] }.combineAll() }
```
Comment (truizlop) [inline Tests/APIHelperTests.swift]: Why do you need to modify the keys?

File: Tests/Utils/Snapshotting+OpenAPI.swift [hunk 4]
```diff
@@ -19,7 +20,7 @@ extension Snapshotting where Value == URL, Format == String {
                                   .unsafeRunSyncEither()
             
             guard either.isRight else {
-                return "error: \(either.leftValue): run bow openAPI in scheme: '\(url.path)', output: '\(directory.path)', template: '\(URL.templates.path)'"
+                return "error: \(either.leftValue). Run bow openAPI using: \n\t - Scheme: '\(url.path)' \n\t - Output: '\(directory.path)' \n\t - Template: '\(URL.templates.path)'"
```
Comment (truizlop) [inline Tests/Utils/Snapshotting+OpenAPI.swift]: Triple quotes strings should make this easier to read.

File: Tests/APIHelperTests.swift (hunk 3 above)
Comment (miguelangel-dev) [reply Tests/APIHelperTests.swift]: some times it assigns (in linux) empty strings... ¯\_(ツ)_/¯

File: Tests/Utils/Snapshotting+OpenAPI.swift (hunk 4 above)
Comment (miguelangel-dev) [reply Tests/Utils/Snapshotting+OpenAPI.swift]: Yep, I will update it

File: Tests/APIHelperTests.swift (hunk 3 above)
Comment (truizlop) [reply Tests/APIHelperTests.swift]: You should be able to fliter out those cases as:
```swift
[String: String].arbitrary.suchThat {
  // Predicate to filter out empty keys
}
```

Comment (miguelangel-dev) [reply Tests/APIHelperTests.swift]: great! I did not know!

**Review by truizlop — APPROVED** [review]

**Pull request merged**